packages/react-components/src/components/utils/GridLayoutUtils.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { BlockProps } from '../GridLayout';
+
+const TARGET_RATIO = 16 / 9;
+const TOLERANCE_RATIO = 8 / 9;
+
+const isRatioBetterThan = (A: number, B: number, targetRatio?: number): boolean => {
+  const _targetRatio = targetRatio ?? TARGET_RATIO;
+  return Math.abs(_targetRatio - A) < Math.abs(_targetRatio - B);
+};
+
+/**
+ * @private
+ */
+export const calculateBlockProps = (n: number, width: number, height: number): BlockProps => {
+  if (width <= 0) {
+    throw Error('Width provided [' + width + '] is less than or equal to 0.');

```

Comment:
The function should not throw errors in the UI. And simply return `numBlocks` as 0 if either height or width is 0.
Since there is no way for a client to handle these errors and they don't serve a purpose.